starter-code/src/viking.js

```diff
@@ -1,11 +1,176 @@
 // Soldier
-class Soldier {}
+class Soldier {
+
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+
+        return this.strength;
+
+    }
+
+    receiveDamage(damage) {
+
+        this.health = this.health - damage;
+
+
+    }
+
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+
+    }
+
+    receiveDamage(damage) {
+
+        this.health = this.health - damage;
+
+        if (this.health > 0) {
+
+            return `${this.name} has received ${damage} points of damage`;
+
+        } else if (this.health <= 0) {
+
+            return `${this.name} has died in act of combat`;
+
+        }
+
+    }
+
+    battleCry() {
+
+        return `Odin Owns You All!`
+
+    }
+}
+
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+
+    receiveDamage(damage) {
+
+        this.health = this.health - damage;
+
+        if (this.health > 0) {
+
+            return `A Saxon has received ${damage} points of damage`;
+
+        } else if (this.health <= 0) {
+
+            return `A Saxon has died in combat`;
+
+        }
+
+
+
+    }
+
+
+}
 
 // War
-class War {}
+class War {
+
+    constructor() {
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+
+
+
+    addViking(viking) {
+
+        this.vikingArmy.push(viking);
+
+    }
+
+    addSaxon(saxon) {
+
+        this.saxonArmy.push(saxon);
+
+    }
+    vikingAttack() {
+
+
+
```

Comment:
Also can be return this.vikingArmy.length > 0 ? this.attack(this.vikingArmy, this.saxonArmy) : `There is no Viking army to attack with`